File: Domain/ProjectWeight.cs
using System.ComponentModel.DataAnnotations.Schema;

namespace Domain
{
    public class ProjectWeight
    {
        public Guid ProjectWeightId { get; set; }
        public Guid ProjectId { get; set; }
        public Project Project { get; set; }
        public DateTime Month { get; set; }

        [Column(TypeName = "decimal(18, 2)")]
        public decimal SalesValueWeight { get; set; }

        [Column(TypeName = "decimal(18, 2)")]
        public decimal ConversionWeight { get; set; }

        [Column(TypeName = "decimal(18, 2)")]
        public decimal RegistrationWeight { get; set; }

        [Column(TypeName = "decimal(18, 2)")]
        public decimal DepositWeight { get; set; }
        public DateTime CreatedAt { get; set; }
        public DateTime UpdatedAt { get; set; }
    }
}

File: client-app/src/app/models/projectWeight.ts
export interface ProjectWeight {
  projectWeightId: string;
  projectId: string;
  projectName?: string;
  month: string;
  salesValueWeight: number;
  conversionWeight: number;
  registrationWeight: number;
  depositWeight: number;
}

export class ProjectWeightFormValues {
  projectWeightId: string = "";
  projectId: string = "";
  month: string = "";
  salesValueWeight: number = 0;
  conversionWeight: number = 0;
  registrationWeight: number = 0;
  depositWeight: number = 0;

  constructor(init?: ProjectWeightFormValues) {
    if (init && init.projectId) {
      this.projectWeightId = init.projectWeightId;
      this.projectId = init.projectId;
      this.month = init.month;
      this.salesValueWeight = init.salesValueWeight;
      this.conversionWeight = init.conversionWeight;
      this.registrationWeight = init.registrationWeight;
      this.depositWeight = init.depositWeight;
    }
  }
}
